src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfig.java

```diff
@@ -127,10 +127,10 @@ private ComplianceConfig(
         this.readEnabledFields = watchedReadFields.stream()
                 .map(watchedReadField -> watchedReadField.split(","))
                 .filter(split -> split.length != 0 && !Strings.isNullOrEmpty(split[0]))
-                .collect(Collectors.toMap(
-                        split -> split[0],
+                .collect(ImmutableMap.toImmutableMap(
+                        split -> WildcardMatcher.from(split[0]),
                         split -> split.length == 1 ?
-                                Collections.singleton("*") : Arrays.stream(split).skip(1).collect(Collectors.toSet())
+                                Collections.singleton("*") : Arrays.stream(split).skip(1).collect(ImmutableSet.toImmutableSet())
```

Comment:
Do you mean inside `getFieldsForIndex()`? I don't think that lazy initialization of `readEnabledFields` is a better choice here as it will require synchronization.